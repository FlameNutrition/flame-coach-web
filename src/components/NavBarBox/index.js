import React from 'react';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from "@material-ui/core";

import PropTypes from 'prop-types';

const useStyles = makeStyles(() => ({
  image: {
    marginRight: '5px',
    width: '24px',
    height: '24px'
  }
}));

const NavBarBox = ({
  title,
  message,
  icon,
  iconEnable,
  btnEnable,
  btnLabel,
  btnHref
}) => {
  const classes = useStyles();

  return (
    <Box
      p={2}
      m={2}
      bgcolor="background.dark"
    >
      <Box
        display="flex"
        justifyContent="center"
      >
        {iconEnable
          ? React.cloneElement(icon, { className: classes.image })
          : null}
        <Typography
          align="center"
          gutterBottom
          variant="h4"
        >
          {title}
        </Typography>
      </Box>
      <Typography
        align="center"
        variant="body2"
      >
        {message}
      </Typography>
      {btnEnable
        ? (
          <Box
            display="flex"
            justifyContent="center"
            mt={2}
          >
            <Button
              color="primary"
              component="a"
              target="_blank"
              href={btnHref}
              variant="contained"
            >
              {btnLabel}
            </Button>
          </Box>
        )
        : null}
    </Box>
  );
};

NavBarBox.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  iconEnable: PropTypes.bool,
  icon: PropTypes.object,
  btnEnable: PropTypes.bool,
  btnLabel: PropTypes.string,
  btnHref: PropTypes.string
};

NavBarBox.defaultProps = {
  btnEnable: false,
  iconEnable: false
};

export default NavBarBox;
