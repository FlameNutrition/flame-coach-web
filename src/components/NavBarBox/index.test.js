import React from 'react';
import { render } from '../../testing/test-utils';
import NavBarBox from './index';

describe('<NavBarBox/ >', () => {
  it('create component with button & icon', () => {
    const { container } = render(
      <NavBarBox
        title="Having issues?"
        message="Please report any issue did you find. Help us to help you!"
        iconEnable
        icon={(
          <img
            alt="ReportIssue"
            src="/static/icons/issue.png"
          />
            )}
        btnEnable
        btnLabel="Report"
        btnHref="mailto:test@gmail.com"
      />
    );

    expect(container).toMatchSnapshot();
    expect(container.querySelector('h4')).toHaveTextContent('Having issues?');
    expect(container.querySelector('p')).toHaveTextContent('Please report any issue did you find. Help us to help you!');
    expect(container.querySelector('span')).toHaveTextContent('Report');
    expect(container.querySelector('a')).toHaveAttribute('href', 'mailto:test@gmail.com');
    expect(container.querySelector('a')).toHaveAttribute('target', '_blank');
    expect(container.querySelector('img')).toHaveAttribute('alt', 'ReportIssue');
    expect(container.querySelector('img')).toHaveAttribute('src', '/static/icons/issue.png');
  });

  it('create component with icon only', () => {
    const { container } = render(
      <NavBarBox
        title="Having issues?"
        message="Please report any issue did you find. Help us to help you!"
        iconEnable
        icon={(
          <img
            alt="ReportIssue"
            src="/static/icons/issue.png"
          />
            )}
      />
    );

    expect(container).toMatchSnapshot();
    expect(container.querySelector('h4')).toHaveTextContent('Having issues?');
    expect(container.querySelector('p')).toHaveTextContent('Please report any issue did you find. Help us to help you!');
    expect(container.querySelector('span')).toBeNull();
    expect(container.querySelector('a')).toBeNull();
    expect(container.querySelector('img')).toHaveAttribute('alt', 'ReportIssue');
    expect(container.querySelector('img')).toHaveAttribute('src', '/static/icons/issue.png');
  });

  it('create component with button only', () => {
    const { container } = render(
      <NavBarBox
        title="Having issues?"
        message="Please report any issue did you find. Help us to help you!"
        btnEnable
        btnLabel="Report"
        btnHref="mailto:test@gmail.com"
      />
    );

    expect(container).toMatchSnapshot();
    expect(container.querySelector('h4')).toHaveTextContent('Having issues?');
    expect(container.querySelector('p')).toHaveTextContent('Please report any issue did you find. Help us to help you!');
    expect(container.querySelector('span')).toHaveTextContent('Report');
    expect(container.querySelector('a')).toHaveAttribute('href', 'mailto:test@gmail.com');
    expect(container.querySelector('a')).toHaveAttribute('target', '_blank');
    expect(container.querySelector('img')).toBeNull();
  });
});
