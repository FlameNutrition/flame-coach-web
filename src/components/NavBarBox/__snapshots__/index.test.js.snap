// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBarBox/ > create component with button & icon 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <img
        alt="ReportIssue"
        class="makeStyles-image-1"
        src="/static/icons/issue.png"
      />
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Having issues?
      </h4>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
    >
      Please report any issue did you find. Help us to help you!
    </p>
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        href="mailto:test@gmail.com"
        tabindex="0"
        target="_blank"
      >
        <span
          class="MuiButton-label"
        >
          Report
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`<NavBarBox/ > create component with button only 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-9"
  >
    <div
      class="MuiBox-root MuiBox-root-10"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Having issues?
      </h4>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
    >
      Please report any issue did you find. Help us to help you!
    </p>
    <div
      class="MuiBox-root MuiBox-root-11"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        href="mailto:test@gmail.com"
        tabindex="0"
        target="_blank"
      >
        <span
          class="MuiButton-label"
        >
          Report
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`<NavBarBox/ > create component with icon only 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <div
      class="MuiBox-root MuiBox-root-7"
    >
      <img
        alt="ReportIssue"
        class="makeStyles-image-5"
        src="/static/icons/issue.png"
      />
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Having issues?
      </h4>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
    >
      Please report any issue did you find. Help us to help you!
    </p>
  </div>
</div>
`;
